### PR TITLE
WRO-4059: Fix lint warnings from updated lint rules.

### PR DESCRIPTION
**Pull request by alexandrumorariu**
Enact-DCO-1.0-Signed-off-by: Alexandru Morariu alexandru.morariu@lgepartner.com

### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Fixed unit tests, fixed ui tests to work async, added comments for disabling lint checks.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-4059.

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

**Comment by daniel-stoian-lgp** [comment]
Note for the next reviewer: Travis fail because it uses cli on develop branch and some new eslitn rules are not defined yet on that branch

File: packages/spotlight/src/tests/spotlight-specs.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ describe('Spotlight', () => {
 			}
 		));
 
-		test('should focus container by node', testScenario(
+		test('should focus container by node with focus on spotlight id', testScenario(
```
Comment (MikyungKim) [inline packages/spotlight/src/tests/spotlight-specs.js:129]:
```suggestion
		test('should focus container by selector', testScenario(
```

File: packages/ui/AnnounceDecorator/tests/useAnnounce-specs.js [hunk 2]
```diff
@@ -42,15 +42,16 @@ describe('useAnnounce', () => {
 	// TODO: find a relevant scenario that works on testing-library
 	test.skip('should set the value passed to announce into the ARIA role="alert" node', () => {
 		const text = '__NOTIFY__';
-		const {container} = render(<Component />);
+		render(<Component data-testid="skipped-test" />);
```
Comment (MikyungKim) [inline packages/ui/AnnounceDecorator/tests/useAnnounce-specs.js:45]:
```suggestion
		render(<Component data-testid="announce-test" />);
```

File: packages/ui/VirtualList/UiVirtualListBase.js [hunk 3]
```diff
@@ -270,7 +270,7 @@ const VirtualListBaseFactory = (type) => {
 		componentDidMount () {
 			if (!this.props.clientSize) {
 				this.calculateMetrics(this.props);
-				this.setState(this.getStatesAndUpdateBounds(this.props));
+				this.setState(this.getStatesAndUpdateBounds(this.props)); // eslint-disable-line react/no-did-mount-set-state
```
Comment (MikyungKim) [inline packages/ui/VirtualList/UiVirtualListBase.js:273]: I don't see any warnings related to this rule. 🤷

File: packages/i18n/tests/ui/specs/Loader/Loader-specs.js [hunk 4]
```diff
@@ -1,89 +1,89 @@
+/* eslint-disable jest/valid-expect */
```
Comment (MikyungKim) [inline packages/i18n/tests/ui/specs/Loader/Loader-specs.js:1]: 🤔 I don't see any warnings related to this from my console.

File: packages/ui/AnnounceDecorator/tests/useAnnounce-specs.js [hunk 5]
```diff
@@ -42,15 +42,16 @@ describe('useAnnounce', () => {
 	// TODO: find a relevant scenario that works on testing-library
 	test.skip('should set the value passed to announce into the ARIA role="alert" node', () => {
 		const text = '__NOTIFY__';
-		const {container} = render(<Component />);
+		render(<Component data-testid="announce-test" />);
 
-		container.find(Base).invoke('announce')(text);
-		container.update();
+		const component = screen.getByTestId('skipped-test');
```
Comment (MikyungKim) [inline packages/ui/AnnounceDecorator/tests/useAnnounce-specs.js:47]: Don't miss this...
```suggestion
		const component = screen.getByTestId('announce-test');
```

File: packages/ui/VirtualList/UiVirtualListBase.js (hunk 3 above)
Comment (alexandrumorariu) [reply packages/ui/VirtualList/UiVirtualListBase.js]: I removed them from UiVirtualListBase and VirtualListBasic.

File: packages/i18n/tests/ui/specs/Loader/Loader-specs.js (hunk 4 above)
Comment (alexandrumorariu) [reply packages/i18n/tests/ui/specs/Loader/Loader-specs.js]: I removed this disable also.

**Pull request merged**